test/sql/test_function/R/test_cast_string_to_datetime

```diff
@@ -0,0 +1,303 @@
+    )), 0) +
+    ifnull(sum(murmur_hash3_32(
+        cast(datetime_str_with_whitespace as datetime)
+    )), 0)
+from t1;
+-- result:
+-1390046567746
+-- !result
+insert into t1
+select 
+    cast(random() *2e9 as int), 
+    null, null, null ,null
+from __row_util;
+-- result:
+-- !result
+    
+select 
+    ifnull(sum(murmur_hash3_32(
+        cast(date_str as datetime)
+    )), 0) +
+    ifnull(sum(murmur_hash3_32(
+        cast(date_str_with_whitespace as datetime)
+    )), 0) +
+    ifnull(sum(murmur_hash3_32(
+        cast(datetime_str as datetime)
+    )), 0) +
+    ifnull(sum(murmur_hash3_32(
+        cast(datetime_str_with_whitespace as datetime)
+    )), 0)
+from t1;
+-- result:
+-1390046567746
+-- !result
+select cast(NULL as datetime);
+-- result:
+None
+-- !result
+select cast("20200101" as datetime);
+-- result:
+2020-01-01 00:00:00
+-- !result
+select cast("20200101010203" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("20200101T010203" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("20200101T0102033" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("20200101T010203.123" as datetime);
+-- result:
+2020-01-01 01:02:03.123000
+-- !result
+select cast("200101" as datetime);
+-- result:
+2020-01-01 00:00:00
+-- !result
+select cast("200101010203" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("200101T010203" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("200101T0102033" as datetime);
+-- result:
+None
+-- !result
+select cast("200101T010203.123" as datetime);
+-- result:
+2020-01-01 01:02:03.123000
+-- !result
+select cast("2020.01.01" as datetime);
+-- result:
+2020-01-01 00:00:00
+-- !result
+select cast("2020.01.01T01.02.03" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("2020.01.01T01.02.033" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("2020.01.01T01.0203.123" as datetime);
+-- result:
+2020-01-01 01:02:03.123000
+-- !result
+select cast("  20200101 " as datetime);
+-- result:
+2020-01-01 00:00:00
+-- !result
+select cast("  20200101010203   " as datetime);
+-- result:
+None
+-- !result
+select cast("  20200101T010203" as datetime);
+-- result:
+2020-01-01 01:02:03
+-- !result
+select cast("20200101T0102033  " as datetime);
+-- result:
+None
+-- !result
+select cast("  20200101T010203.123    \n" as datetime);
+-- result:
+2020-01-01 01:02:03.123000
+-- !result
+select cast("  20200101T010203.123    \n \t \v \f \r " as datetime);
+-- result:
+2020-01-01 01:02:03.123000
+-- !result
+    
+select cast("2020.13.29" as datetime);
+-- result:
+None
+-- !result
+select cast("2020.13.61" as datetime);
+-- result:
+None
+-- !result
+select cast("2020.02.29" as datetime);
+-- result:
+2020-02-29 00:00:00
+-- !result
+select cast("2020.02.28" as datetime);
+-- result:
+2020-02-28 00:00:00
+-- !result
+select cast("2000.02.29" as datetime);
```

Comment:
Does the const cast process in FE or BE?

Reply:
Yes, for `select cast("2000.02.29" as datetime)`, but for `select cast('2020-02-29' as datetime)` it is calculated in FE.